packages/rich-text/src/plugins/Hr/index.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as Slate from 'slate-react';
+import { css, cx } from 'emotion';
+import tokens from '@contentful/forma-36-tokens';
+import { EditorToolbarButton } from '@contentful/forma-36-react-components';
+import { Transforms, Editor } from 'slate';
+import { BLOCKS } from '@contentful/rich-text-types';
+import { useCustomEditor } from '../../hooks/useCustomEditor';
+import { CustomElement } from 'types';
+
+const styles = {
+  container: css`
+    margin: 0 0 ${tokens.spacingL};
+  `,
+  hr: css`
+    margin: 0;
+    height: ${tokens.spacingM};
+    background: transparent;
+    position: relative;
+    border: 0;
+    user-select: none;
+    &:hover {
+      cursor: pointer;
+    }
+    &::after {
+      content: '';
+      position: absolute;
+      width: 100%;
+      height: 1px;
+      background: ${tokens.colorElementMid};
+      top: 50%;
+    }
+  `,
+  hrSelected: css`
+    &::after {
+      background: ${tokens.colorPrimary};
+      -webkit-box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+      box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+    }
+  `,
+};
+
+interface ToolbarHrButtonProps {
+  isDisabled?: boolean;
+}
+
+export function withHrEvents(editor, event) {
+  const [currentFragment] = Editor.fragment(editor, editor.selection.focus.path) as CustomElement[];
+
+  if (
+    event.keyCode === 13 &&
+    Editor.isVoid(editor, currentFragment) &&
```

Comment:
Why do we need to asset whether the the current fragment is void? Are there cases where the `type` would be `HR` but not void, and that would determine the outcome?

Reply:
You are maybe right, if we have `currentFragment.type === BLOCKS.HR` we might not need this `Editor.isVoid` checking.